src/runtime/cpu/kernels/kernels.cpp

```diff
@@ -1,25 +1,29 @@
 #include "runtime/cpu/kernels/kernels.hpp"
 
-void deepworks::CPULinearForward(ConstMatrix X, ConstMatrix W, Matrix result) {
+void deepworks::CPUMultiply(ConstMatrix X1, ConstMatrix X2, Matrix result) {
```

Comment:
Not sure, that `CPUMultiply` is a good name. Often Multiply means elementwise multiplication. I think `CPUMatMul` is more clear.

Reply:
I'm not that it's matrix multiplication, seems just elementwise

Reply:
I'll double check ofc

Reply:
Left `CPULInearForward` so far